backend/routes/chart.py

```diff
@@ -74,37 +78,81 @@ def get_aggregated_pie_chart_data(req: Request,
     return value
 
 
-@chart_route.get("/chart/jmp-data/{form_id:path}/{question_id:path}",
-                 name="charts:get_aggregated_jmp_chart_data",
-                 summary="get jmp chart aggregate data",
-                 tags=["Charts"])
-def get_aggregated_jmp_chart_data(req: Request,
-                                  form_id: int,
-                                  question_id: int,
-                                  administration: Optional[int] = None,
-                                  q: Optional[List[str]] = Query(None),
-                                  session: Session = Depends(get_session)):
-    options = check_query(q) if q else None
+def group_children(p, data_source, labels):
+    data = list(
+        filter(lambda d: (d["administration"] in p["children"]), data_source)
+    )
+    data = [{"category": d["category"], "data": d["data"]} for d in data]
+    total = len(data)
+    childs = []
+    groups = groupby(data, key=lambda d: d["category"])
+    counter = defaultdict()
+    for k, values in groups:
+        for v in list(values):
+            if v["category"] in list(counter):
+                counter[v["category"]] += 1
+            else:
+                counter[v["category"]] = 1
+    for lb in labels:
```

Comment:
Hi mas @dedenbangkit , I've fixed the percentages and added the correct colors and labels coming from the .category.json configuration